msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java

```diff
@@ -0,0 +1,179 @@
+        // pca should be in MULTIPLE account mode starting out
+        Assert.assertTrue(mApplication instanceof MultipleAccountPublicClientApplication);
+
+        //we should NOT be in shared device mode
+        Assert.assertFalse(mApplication.isSharedDevice());
+
+        // perform shared device registration
+        mBroker.performSharedDeviceRegistration(
+                mLoginHint, LabConfig.getCurrentLabConfig().getLabUserPassword()
+        );
+
+        // re-create PCA after device registration
+        mApplication = PublicClientApplication.create(mContext, getConfigFileResourceId());
+
+        // pca should now be in SINGLE account mode
+        Assert.assertTrue(mApplication instanceof SingleAccountPublicClientApplication);
+
+        // we should be in shared device mode
+        Assert.assertTrue(mApplication.isSharedDevice());
+
+        //creating a basic temp user account
+        final String newUser = LabConstants.TempUserType.BASIC;
+        final String username = LabUserHelper.loadTempUser(newUser);
+        String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(30));
+
+
+        final SingleAccountPublicClientApplication singleAccountPCA =
+                (SingleAccountPublicClientApplication) mApplication;
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        // try sign in with an account from the same tenant
+        singleAccountPCA.signIn(mActivity, username, mScopes, successfulInteractiveCallback(latch));
+
+        final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                .loginHint(username)
+                .sessionExpected(false)
+                .consentPageExpected(false)
+                .broker(mBroker)
+                .prompt(PromptParameter.SELECT_ACCOUNT)
+                .expectingBrokerAccountChooserActivity(false)
+                .build();
+
+        AadPromptHandler aadPromptHandler = new AadPromptHandler(promptHandlerParameters);
+        aadPromptHandler.handlePrompt(username, password);
+
+        latch.await();
+
+        //launching azure sample app and confirming user signed in or not.
+        final AzureSampleApp azureSampleApp = new AzureSampleApp();
+        azureSampleApp.install();
+        azureSampleApp.launch();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        azureSampleApp.confirmSignedIn(username);
+
+        //signing in chrome browser.
+        IBrowser chrome = new BrowserChrome();
+        chrome.launch();
+        chrome.handleFirstRun();
+        chrome.clear();
+
+        //relaunching chrome after clearing history of chrome.
+        chrome.launch();
+        chrome.handleFirstRun();
+        chrome.navigateTo("https://myapps.microsoft.com");
+        UiAutomatorUtils.handleInput("i0116", username);
+        UiObject button = UiAutomatorUtils.obtainUiObjectWithText("Next");
```

Comment:
Let's use specific names like nextButton

```suggestion
        UiObject nextButton = UiAutomatorUtils.obtainUiObjectWithText("Next");
```

Reply:
changed button to nextButton